xml-jaxp/src/main/java/org/jboss/as/quickstart/xml/jaxp/FileUploadBean.java

```
@@ -0,0 +1,137 @@
+     * 
+     * @return
+     */
+    public String getFactoryName() {
+        return factoryName;
+    }
+
+    /**
+     * Setter for selection field name.
+     * 
+     * @param inputName
+     */
+
+    public void setFactoryName(String factoryName) {
+        this.factoryName = factoryName;
+    }
+
+    /**
+     * Getter for XMLParserFactory class name. Value must be a valid implementation of XMLParserFactory interface
+     * 
+     * @return
+     */
+    public String getParserFactoryClass() {
+        return parserFactoryClass;
+    }
+
+    /**
+     * Setter for XMLParserFactory class name.
+     * 
+     * @param parserFactoryClass
+     */
+    public void setParserFactoryClass(String parserFactoryClass) {
+        this.parserFactoryClass = parserFactoryClass;
+    }
+
+    /**
+     * Getter for books catalog.
+     * 
+     * @return
+     */
+    public Catalog getCatalog() {
+        return catalog;
+    }
+
+    /**
+     * Setter for books catalog.
+     * 
+     * @param catalog
+     */
+    public void setCatalog(Catalog catalog) {
+        this.catalog = catalog;
+    }
+
+    /**
+     * Action method invoked from UploadServlet once it parses request with 'multipart/form-data' form data and fetches uploaded
+     * file.<br>
+     * This method expects file to be passed as {@link InputStream}. It uses provided {@link #parserFactoryClass} to
+     * instantiate parser factory and create parser instance, to be able to parse passed file. Result is stored in
+     * {@link #catalog} instance variable.
+     * 
+     * @param stringBuffer
+     */
+    @SuppressWarnings("static-access")
+    public void parseUpload(InputStream is) {
+        try {
+            this.parserFactory = (XMLParserFactory) this.getClass().forName(this.getParserFactoryClass()).newInstance();
+            this.catalog = this.parserFactory.instance().parse(is);
+        } catch (Exception e) {
+            // TODO Auto-generated catch block
```

Do you really want to swallow this exception? Shouldn't it be wrapped in RuntimeException (or the like) and re-thrown?